src/logger.rs

```diff
@@ -0,0 +1,109 @@
+use log::Level;
+use settings::Logging;
+
+pub fn flush() {
+    // flushing `println!` shouldn't be nessasary unless we are writing
+    // to a hard fd or line discipline is fully buffered. i've
+    // implemented `flush` anyway to this to ensure no termios emit bugs
+    use std::io::stdout;
+    use std::io::Write;
+    let _ = stdout().flush();
+}
+
+mod raw_logger {
+    use log::{set_boxed_logger, set_max_level, Level, Log, Metadata, Record, SetLoggerError};
+
+    struct SimpleLogger {
+        name: String,
+        level: Level,
+    }
+
+    impl Log for SimpleLogger {
+        fn enabled(&self, metadata: &Metadata) -> bool {
+            metadata.level() <= self.level
+        }
+
+        fn log(&self, record: &Record) {
+            if self.enabled(record.metadata()) {
+                println!(
```

Comment:
Think we should be logging to stderr instead, not overly opinionated on that but I believe thats the typical convention.